app/lib/methods/actions.js

```diff
@@ -0,0 +1,142 @@
+import random from '../../utils/random';
+import EventEmitter from '../../utils/events';
+import Navigation from '../Navigation';
+import { showErrorAlert } from '../../utils/info';
+import I18n from '../../i18n';
+
+const TRIGGER_TIMEOUT = 5000;
+
+const ACTION_TYPES = {
+	ACTION: 'blockAction',
+	SUBMIT: 'viewSubmit',
+	CLOSED: 'viewClosed'
+};
+
+export const MODAL_ACTIONS = {
+	MODAL: 'modal',
+	OPEN: 'modal.open',
+	CLOSE: 'modal.close',
+	UPDATE: 'modal.update',
+	ERRORS: 'errors'
+};
+
+const triggersId = new Map();
+
+const invalidateTriggerId = (id) => {
+	const appId = triggersId.get(id);
+	triggersId.delete(id);
+	return appId;
+};
+
+export const generateTriggerId = (appId) => {
+	const triggerId = random(17);
+	triggersId.set(triggerId, appId);
+	setTimeout(invalidateTriggerId, TRIGGER_TIMEOUT, triggerId);
+	return triggerId;
+};
+
+export const handlePayloadUserInteraction = (type, { triggerId, ...data }) => {
+	if (!triggersId.has(triggerId)) {
+		return;
+	}
+
+	const appId = invalidateTriggerId(triggerId);
+	if (!appId) {
+		return;
+	}
+
+	// TODO not sure this will always have 'view.id'
```

Comment:
```suggestion
	// TODO: not sure this will always have 'view.id'
```

Just to editors understand it as todo